From 8e54d6d12154a1e10abb314ef44759f82713845a Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:42 -0400
Subject: [tools] lib/traceevent: Unregister handler when mac80211 plugin is unloaded

Message-id: <1408953210-26343-275-git-send-email-jolsa@redhat.com>
Patchwork-id: 89033
O-Subject: [PATCH RHEL7.1 BZ1133083 274/322] tools lib traceevent: Unregister handler when mac80211 plugin is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit ec7c6debdd446ad2262f236d13964efae90ba0f7
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:14 2014 +0900

description
===========
The event handler should be unregistered when the plugin is unloaded
otherwise it'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_mac80211.c b/tools/lib/traceevent/plugin_mac80211.c
index 558a3b9..7e15a0f 100644
--- a/tools/lib/traceevent/plugin_mac80211.c
+++ b/tools/lib/traceevent/plugin_mac80211.c
@@ -93,3 +93,10 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
           drv_bss_info_changed, NULL);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_event_handler(pevent, -1, "mac80211",
+     "drv_bss_info_changed",
+     drv_bss_info_changed, NULL);
+}
-- 
1.7.1